### ISSUE TITLE
Scripting events are not working on 64-bit build

### ISSUE THREAD

**Issue by helloer** [opensoldat/opensoldat#60]
Following script code works flawlessly on a 32-bit soldatserver but on a 64-bit build any assigned game event is not called up. This seems to be related to https://github.com/remobjects/pascalscript/issues/153 and [this](https://stackoverflow.com/questions/33648962/compiling-pascalscript-in-64-bit-mode-events-not-working).

```pascal
procedure MyOnClockTickHandler(Ticks: Integer); // this won't be called on 64bit
begin
  WriteLn('OnClockTick event called: ' + IntToStr(Ticks));
end;

begin
  WriteLn('This works');
  Game.OnClockTick := @MyOnClockTickHandler;
end.
```

**Comment by Dankage102** [comment]
Maybe this might be one of the reasons https://wiki.freepascal.org/Integer 

> For 64-bit calculation, FPC does support 64-bit Int64 though, which may be defined as a value of -263 through 263-1, or -9 223 372 036 854 775 808 ... 9 223 372 036 854 775 807.
> 
> This will cause code expecting an integer and a pointer to be the same size to fail as a 64bit machine uses 64-bit pointers. To allow you to write portable code, the FPC system unit introduces the types PtrInt and PtrUInt which are signed and unsigned integer data types with the same size as a pointer.

**Comment by BranDougherty** [comment]
Thanks for the links and minimal example. I consider this a high priority issue but seeing as I am not very familiar with ScriptCore it will likely take me a while to dig into this.

**Comment by helloer** [comment]
Updating pascalscript to latest revision, applying https://github.com/ccy/pascalscript/commit/1013b4f7deb549808f21312fa3ddd722170e6892 and commenting out `{$define empty_methods_handler}` in `uPSRuntime.pas` _seems_ to fix the problem on win64 but on linux the problem still occurs since the patch is for windows only.

**Comment by BranDougherty** [comment]
I was able to get this working on Linux 64 bit, at least on my machine (Arch Linux, FPC 3.2.2)... I checked the disassembly of  `OnClockTickWriteHelper` and it expects 3 parameters in registers (Self of ScriptGame in rdi, code pointer of callback in rsi, and data pointer of callback in rdx). However PascalScript is always passing the procedure pointer (called a btProcPtr internally) on the stack. So I replaced the StoreStack call in x64.inc with 2 StoreReg calls and it worked.

In the end what I had to do wasn't super complicated but it took me a very long time to figure it out... I'd also be surprised if this is all we need to do, so I'll make a draft PR in a little bit and we can test it.
